msolve 0.9.0 fails to build from source on debian-bullseye-i386

```
  [msolve-0.9.0]   [spkg-install] libtool: link: gcc -mmmx -msse -msse2 -msse3 -mssse3 -msse4.1 -msse4.2 -msse4a -msha -maes -mavx -mfma -mavx2 -mrdrnd -mbmi -mbmi2 -madx -mabm -fopenmp -g -O2 -Wl,-rpath-link -Wl,/sage/local/lib -Wl,-rpath -Wl,/sage/local/lib -Wl,-rpath-link -Wl,/sage/local/lib -Wl,-rpath -Wl,/sage/local/lib -o .libs/msolve src/msolve/main.o  -L/sage/local/lib src/neogb/.libs/libneogb.so src/fglm/.libs/libfglm.a src/usolve/.libs/libusolve.a -lflint -lmpfr -lgmp -lm -fopenmp -Wl,-rpath -Wl,/sage/local/lib
  [msolve-0.9.0]   [spkg-install] /usr/bin/ld: src/fglm/.libs/libfglm.a(libfglm_la-fglm_core.o): in function `_mm256_hsum':
  [msolve-0.9.0]   [spkg-install] /sage/local/var/tmp/sage/build/msolve-0.9.0/src/src/fglm/linalg-fglm.c:211: undefined reference to `_mm_cvtsi128_si64'
  [msolve-0.9.0]   [spkg-install] /usr/bin/ld: /sage/local/var/tmp/sage/build/msolve-0.9.0/src/src/fglm/linalg-fglm.c:211: undefined reference to `_mm_cvtsi128_si64'
  [msolve-0.9.0]   [spkg-install] /usr/bin/ld: /sage/local/var/tmp/sage/build/msolve-0.9.0/src/src/fglm/linalg-fglm.c:211: undefined reference to `_mm_cvtsi128_si64'
  [msolve-0.9.0]   [spkg-install] /usr/bin/ld: /sage/local/var/tmp/sage/build/msolve-0.9.0/src/src/fglm/linalg-fglm.c:211: undefined reference to `_mm_cvtsi128_si64'
  [msolve-0.9.0]   [spkg-install] /usr/bin/ld: /sage/local/var/tmp/sage/build/msolve-0.9.0/src/src/fglm/linalg-fglm.c:211: undefined reference to `_mm_cvtsi128_si64'
  [msolve-0.9.0]   [spkg-install] /usr/bin/ld: src/fglm/.libs/libfglm.a(libfglm_la-fglm_core.o):/sage/local/var/tmp/sage/build/msolve-0.9.0/src/src/fglm/linalg-fglm.c:211: more undefined references to `_mm_cvtsi128_si64' follow
```
https://github.com/passagemath/passagemath/actions/runs/16405589442/job/46351032593#step:11:2851

--- a/src/fglm/linalg-fglm.c
+++ b/src/fglm/linalg-fglm.c
@@ -211,7 +211,13 @@ uint64_t _mm256_hsum(__m256i a)
     __m256i sum_lo = _mm256_add_epi64(a, a_hi);
     __m128i sum_hi = _mm256_extracti128_si256(sum_lo, 1);
     __m128i sum = _mm_add_epi64(_mm256_castsi256_si128(sum_lo), sum_hi);
+#if defined(__x86_64__) || defined(_M_X64)
     return (uint64_t) _mm_cvtsi128_si64(sum);
+#else
+    uint64_t lo = (uint32_t) _mm_cvtsi128_si32(sum);
+    uint64_t hi = (uint32_t) _mm_cvtsi128_si32(_mm_srli_si128(sum, 4));
+    return lo | (hi << 32);
+#endif
 }
 
 uint32_t _nmod32_vec_dot_split_avx2(const uint32_t * vec1, const uint32_t * vec2, int64_t len,

